lib/charliecloud.py

```diff
@@ -701,6 +701,19 @@ def log(msg, hint, trace, color, prefix, end="\n"):
    if (color is not None):
       color_reset(log_fp)
 
+def monkey_write_streams():
+   """Monkey patch to replace problematic characters in stdout and stderr
+      streams when running Python 3.6. (see #1629)."""
+   def monkey_write_insert(f):
+      write_orig = f.write
+      def write_monkey(text):
+         text = text.replace("“", "\"").replace("”", "\"").replace("’", "'")
+         write_orig(text)
+      f.write = write_monkey
+   if (sys.version_info[:2] <= (3, 6)):
+      monkey_write_insert(sys.stdout)
+      monkey_write_insert(sys.stderr)
```

Comment:
We'll want to also trigger the fix on the character set not being supported somehow; perhaps locale? terminal something? 3.11 seems to know that the terminal supports Unicode characters despite the locale being `C`.

It's probably possible to directly test if the characters can be encoded correctly. I assume lots of encodings have the curly quotes, not just UTF-8. You could maybe get the default encoding somehow, try to encode a test string, and install the workaround if that throws an exception. This would have the advantage of evaluating results rather than trying to guess what work would based on terminal/locale/etc.